app/services/cocina_generator/structural/generator.rb

```diff
@@ -24,38 +26,63 @@ def generate
       end
 
       # 1) Start with the existing filesets from SDR
-      # 1) Only keep the filesets that have attached_files that are preserved
-      # 2) Add new filesets that have attached_files in the local store.
+      # 2) Rebuild filesets that have attached_files that are preserved
+      # 3) Add new filesets that have attached_files in the local store.
       def build_filesets
         filesets = find_preserved_filesets
-        filesets + work_version.staged_files.map.with_index(filesets.size + 1) { |af, n| build_fileset(af, n) }
+        # rebuild the the ones that are still preserved
+        filesets.each do |filename, fileset|
+          rebuilt_fileset = build_fileset(attached_file: get_preserved_file(filename), fileset:)
+          filesets[filename] = rebuilt_fileset
+        end
```

Comment:
It looks like this snippet is iterating over a hash and updating each value? If so, I wonder about using a different iteration strategy is worth it. E.g.:

```suggestion
        filesets = find_preserved_filesets.to_h do |filename, fileset| 
          [
            filename, 
            build_fileset(attached_file: get_preserved_file(filename), fileset:)
          ] 
        end 
```

Reply:
I wasn't familiar with that iteration approach, thank you! It is more concise. Why is it that you have to add `.to_h` on an object that is already a hash for it to work?

Reply:
This is not obvious at all: when you send `#to_h` to a hash, it returns itself (which seems pointless)... but if a block is also passed, the return value of the block for each key/value pair will be used as the new key/value pair.

So, you can think of `hash.to_h { |key, value| [key, value] }` as a short form of `hash.map { |key, value| [key, value] }.to_h`